### PR TITLE
add support for ADPF

### PR DESCRIPTION
**Pull request by pixelflinger**
- we add a new `utils::PerformanceHintManager` class that is only available on Android
- we also make use `utils::PerformanceHintManager` in `PlatformEGLAndroid`

TODOs: 
- add support in the Vulkan backend
- add support for the main thread and thread pools

### CODE REVIEW

File: filament/src/details/Renderer.cpp [hunk 1]
```diff
@@ -298,7 +298,10 @@ bool FRenderer::beginFrame(FSwapChain* swapChain, uint64_t vsyncSteadyClockTimeN
         FEngine& engine = mEngine;
         FEngine::DriverApi& driver = engine.getDriverApi();
 
-        driver.beginFrame(appVsync.time_since_epoch().count(), mFrameId);
+        driver.beginFrame(
+                appVsync.time_since_epoch().count(),
+                int64_t(1'000'000'000.0 / mDisplayInfo.refreshRate),
```
Comment (poweifeng) [inline filament/src/details/Renderer.cpp:303]: [comment] mentions it that `refreshRate` is `0` for offscreen, is the comment outdated? 

[comment]: https://github.com/google/filament/blob/main/filament/include/filament/Renderer.h#L77

File: filament/backend/src/opengl/platforms/PlatformEGLAndroid.cpp [hunk 2]
```diff
@@ -112,8 +117,34 @@ void PlatformEGLAndroid::terminate() noexcept {
...
+    PlatformEGL::beginFrame(monotonic_clock_ns, refreshIntervalNs, frameId);
+}
+
+void backend::PlatformEGLAndroid::preCommit() noexcept {
+    if (mPerformanceHintSession.isValid()) {
+        auto const actualWorkDuration = std::chrono::duration_cast<std::chrono::nanoseconds>(
+                clock::now() - mStartTimeOfActualWork);
+        mPerformanceHintSession.reportActualWorkDuration(actualWorkDuration.count());
+    }
+    OpenGLPlatform::preCommit();
```
Comment (poweifeng) [inline filament/backend/src/opengl/platforms/PlatformEGLAndroid.cpp:137]: `PlatformEGL` here?

File: filament/src/details/Renderer.cpp (hunk 1 above)
Comment (pixelflinger) [reply filament/src/details/Renderer.cpp:303]: ah good catch. I should probably pass 0 in that case.

File: filament/backend/src/opengl/platforms/PlatformEGLAndroid.cpp (hunk 2 above)
Comment (pixelflinger) [reply filament/backend/src/opengl/platforms/PlatformEGLAndroid.cpp:137]: uh. yes thanks!

**Review by poweifeng — APPROVED** [review]

**Review by bejado — APPROVED** [review]

**Pull request merged**